transform-plugins/src/main/java/co/cask/hydrator/plugin/Compressor.java

```diff
@@ -0,0 +1,267 @@
+    try {
+      Schema outputSchema = Schema.parseJson(config.schema);
+      List<Field> outFields = outputSchema.getFields();
+      for(Field field : outFields) {
+        outSchemaMap.put(field.getName(), field.getSchema().getType());
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+  }
+  
+
+  @Override
+  public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) throws Exception {
+    StructuredRecord.Builder builder = StructuredRecord.builder(outSchema);
+    
+    Schema inSchema = in.getSchema();
+    List<Field> inFields = inSchema.getFields();
+    
+    // Iterate through input fields. Check if field name is present 
+    // in the fields that need to be encoded, if it's not then write 
+    // to output as it is. 
+    for(Field field : inFields) {
+      LOG.info("Looking at field '" + field.getName() + "'");
+      String name = field.getName();
+      
+      // Check if output schema also have the same field name. If it's not 
+      // then continue.
+      if(!outSchemaMap.containsKey(name)) {
+        continue;
+      }
+      
+      Schema.Type outFieldType = outSchemaMap.get(name);
+      
+      // Check if the input field name is configured to be encoded. If the field is not 
+      // present or is defined as none, then pass through the field as is. 
+      if(!compMap.containsKey(name) || compMap.get(name) == CompressorType.NONE) {
+        builder.set(name, in.get(name));          
+      } else {
+        // Now, the input field could be of type String or byte[], so transform everything
+        // to byte[] 
+        byte[] obj = new byte[0];
+        if (field.getSchema().getType() == Schema.Type.BYTES){
+          obj = in.get(name);
+        } else if (field.getSchema().getType() == Schema.Type.STRING) {
+          obj = ((String)in.get(name)).getBytes();
+        }
+        
+        // Now, based on the encode type configured for the field - encode the byte[] of the 
+        // value.
+        byte[] outValue = new byte[0];
+        CompressorType type = compMap.get(name);
+        if(type == CompressorType.SNAPPY) {
+          outValue = Snappy.compress(obj);
+        } else if (type == CompressorType.ZIP) {
+          outValue = compressZIP(obj);
+        } else if (type == CompressorType.GZIP) {
+          outValue = compressGZIP(obj);
+        }
+        
+        // Depending on the output field type, either convert it to 
+        // Bytes or to String. 
+        if(outFieldType == Schema.Type.BYTES) {
+          builder.set(name, outValue);
+        }
+      }
+    }
+    emitter.emit(builder.build());
+  }
+
+  private static byte[] compressGZIP(byte[] input) throws IOException {
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    GZIPOutputStream gzip = new GZIPOutputStream(out);
+    gzip.write(input, 0, input.length);
```

Comment:
Since write throws an IOException, we should close in a finally block. Same for other compressors
```
try {
  gzip.write(input, 0, input.length);
} finally {
  gzip.close();
}
```